plugins/Carbonix/UI/EmergencyTab.cs

```diff
@@ -124,5 +124,52 @@ private void but_fenceenable_Click(object sender, EventArgs e)
             // Send aux function of 11:FENCE to high
             Host.comPort.doCommand(MAVLink.MAV_CMD.DO_AUX_FUNCTION, 11, 2, 0, 0, 0, 0, 0);
         }
+
+        private void but_arm_Click(object sender, EventArgs e)
+        {
+            arm_disarm(true);
+        }
+        private void but_disarm_Click(object sender, EventArgs e)
+        {
+            arm_disarm(false);
+        }
+
+        private void arm_disarm(bool arm)
+        {
+            if (!Host.comPort.BaseStream.IsOpen)
+                return;
+
+            // arm the MAV
+            try
+            {
+                var action = arm ? "Arm" : "Disarm";

```

Comment:
only one button is active (enabled) at any given time?
[but_arm.enabled == true if but_disarm.enabled == false]

Reply:
Both buttons should be active all the time (if I added enable/disable logic somewhere, please point it out and I'll fix). The `arm` variable here is a boolean input argument to the `arm_disarm` function, not tied to the enabled flag of a button.